src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -180,64 +165,56 @@ public String getFilename() {
                 .buildAndExpand(networkUuid, voltageLevelId)
                 .toUriString();
 
-        ResponseEntity<byte[]> responseEntity = singleLineDiagramServerRest.exchange(path,
-                HttpMethod.GET,
-                HttpEntity.EMPTY,
-                byte[].class);
-        return responseEntity.getBody();
+        return webClient.get()
+                .uri(singleLineDiagramServerBaseUri + path)
+                .retrieve()
+                .bodyToMono(byte[].class);
     }
 
-    String getVoltageLevelSvgAndMetadata(UUID networkUuid, String voltageLevelId, boolean useName, boolean centerLabel, boolean diagonalLabel,
-                                         boolean topologicalColoring) {
-        String path = UriComponentsBuilder.fromPath("/" + SINGLE_LINE_DIAGRAM_API_VERSION + "/svg-and-metadata/{networkUuid}/{voltageLevelId}")
+    Mono<String> getVoltageLevelSvgAndMetadata(UUID networkUuid, String voltageLevelId, boolean useName, boolean centerLabel, boolean diagonalLabel,
+                                               boolean topologicalColoring) {
+        String path = UriComponentsBuilder.fromPath(DELIMITER + SINGLE_LINE_DIAGRAM_API_VERSION + "/svg-and-metadata/{networkUuid}/{voltageLevelId}")
                 .queryParam("useName", useName)
                 .queryParam("centerLabel", centerLabel)
                 .queryParam("diagonalLabel", diagonalLabel)
                 .queryParam("topologicalColoring", topologicalColoring)
                 .buildAndExpand(networkUuid, voltageLevelId)
                 .toUriString();
 
-        ResponseEntity<String> responseEntity = singleLineDiagramServerRest.exchange(path,
-                HttpMethod.GET,
-                HttpEntity.EMPTY,
-                String.class);
-        return responseEntity.getBody();
+        return webClient.get()
+                .uri(singleLineDiagramServerBaseUri + path)
+                .retrieve()
+                .bodyToMono(String.class);
     }
 
-    private NetworkInfos persistentStore(UUID caseUuid) {
-        String path = UriComponentsBuilder.fromPath("/" + NETWORK_CONVERSION_API_VERSION + "/networks")
+    private Mono<NetworkInfos> persistentStore(UUID caseUuid) {
+        String path = UriComponentsBuilder.fromPath(DELIMITER + NETWORK_CONVERSION_API_VERSION + "/networks")
                 .queryParam(CASE_UUID, caseUuid)
                 .buildAndExpand()
                 .toUriString();
 
-        ResponseEntity<NetworkInfos> responseEntity = networkConversionServerRest.exchange(path,
-                HttpMethod.POST,
-                HttpEntity.EMPTY,
-                NetworkInfos.class);
-        return responseEntity.getBody();
+        return webClient.post()
+                .uri(networkConversionServerBaseUri + path)
+                .retrieve()
+                .bodyToMono(NetworkInfos.class);
     }
 
     List<VoltageLevelAttributes> getNetworkVoltageLevels(UUID networkUuid) {
```

Comment:
should we convert this to a nonblocking call ?

Reply:
of course !